bravado/mapping/model.py

```diff
@@ -144,3 +154,49 @@ def create_flat_dict(model):
         else:
             model_dict[k] = create_flat_dict(v)
     return model_dict
+
+
+def tag_models(spec_dict):
+    # TODO: unit test + docstring
+    # Tag #/definitions as being models with a 'x-model' key so that they can
+    # be recognized after jsonref inlines $refs
+    models_dict = spec_dict.get('definitions', {})
+    for model_name, model_spec in models_dict.iteritems():
+        model_type = model_spec.get('type')
+
+        # default type type to 'object' since most swagger specs don't bother
+        # to specify this
+        if model_type is None:
+            model_type = model_spec['type'] = 'object'
+
+        # only tag objects. Not all #/definitions map to a Model type - can
+        # be primitive or array, for example
+        if model_type == 'object':
+            model_spec[MODEL_MARKER] = model_name
+
+
+def fix_malformed_model_refs(spec):
+    """jsonref doesn't understand  { $ref: Category } so just fix it up to
+    { $ref: #/definitions/Category } when the ref name matches a #/definitions
+    name. Yes, this is hacky!
```

Comment:
I think we're better off doing this in a fork on jsonref (and see about getting it upstream) than inline in bravado.  

Maybe a TODO for now?